jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/ee/rs/core/headers/HttpHeadersTest.java

```diff
@@ -67,6 +70,39 @@ public String headersGet() {
     return sb.toString();
   }
 
+  @GET
+  @Path("/contains-headers")
+  public String containsHeadersGet() {
+    sb = new StringBuffer();
+    List<String> myHeaders = Arrays.asList("Accept", "Content-Type", "Header3");
+
+    try {
+        assertTrue(hs.containsHeaderString("accept", "text/html"::equalsIgnoreCase));
+        sb.append("Header: accept contains text/html; ");
+      
+        //Verify Predicate and separator character usage
+        assertTrue(hs.containsHeaderString("Accept", ",", "Text/html;Level=1"::equalsIgnoreCase));
+        sb.append("Header: accept contains text/html;level=1; ");
+        
+        //Verify incorrect separator character fails
+        assertFalse(hs.containsHeaderString("Accept", ";", "text/html;level=1"::equals));
+        sb.append("Incorrect separator character fails as expected; ");
+        
+        //Verify white space in parameter not trimmed
+        assertFalse(hs.containsHeaderString("header3", "**", "value3"::equalsIgnoreCase));
+        sb.append("White space not trimmed from parameter as expected; ");
+        
+        //Verify white space in front and back of value trimmed
+        assertTrue(hs.containsHeaderString("HEADER3", "**", "value2"::equalsIgnoreCase));
```

Comment:
Same here, perhaps star is not the best reg exp separator.

Reply:
Agreed